### PR TITLE
Not all valid yaml files start with ---

### PR DESCRIPTION
**Pull request by strets123**
The yaml spec says that a valid yaml file "may" start with --- see
http://yaml.org/spec/1.0/#id2561718

Added a test using yaml refrences and generated yaml similar to the json
one.

### CODE REVIEW

File: dollar_ref/__init__.py [hunk 1]
```diff
@@ -191,7 +191,8 @@ def read_file(path: str) -> dict:
         raw = file.read()
 
         try:
-            if raw.startswith('---'):
+            if raw.startswith('---') or any(
+                    [path.lower().endswith(s) for s in ['.yaml', '.yml']]):
```
Comment (bagrat) [inline dollar_ref/__init__.py]: *minor:* would be great to have some brief comment on this, just like you described in the PR description.

**Review by bagrat — APPROVED** [review]
First thanks for the PR! Looks good 👍 added a minor comment, if you feel like you can add it just go ahead and push another commit, otherwise I can merge this and add it later.

**Comment by strets123** [comment]
I have added the comment.

Thanks for merging!

File: dollar_ref/__init__.py [hunk 2]
```diff
@@ -184,26 +184,36 @@ def read_file(path: str) -> dict:
 
     This function automatically detects whether the file is a JSON or YAML
     and decodes accordingly.
+
+    Detection is based on the filename with files ending in .yml or yaml
+    loaded as yaml. Yaml file contents may begin with ---
+    http://yaml.org/spec/1.0/#id2561718 therefore
+    this is another possible criterion. Any data not fitting these criteria
+    is parsed as JSON.
```
Comment (bagrat) [inline dollar_ref/__init__.py:192]: Perfect!

**Comment by bagrat** [comment]
FYI @strets123, just released `v0.1.1` that contains your fix.

**Pull request merged**